[BUG] Building Problem : 'int freeMemory()' defined but not used [-Wunused-function]

### Bug Description

Building problem when build the today bugfix 2.0 for the SKR mini E3 V2.0
The problem is : 'int freeMemory()' defined but not used [-Wunused-function]

### Steps to Reproduce

1. Download the last bugfix 2.0
2. Download the last bugfix or import configuration files (same problem for both)
4. Paste the config files from config\examples\Creality\Ender-3 Pro\BigTreeTech SKR Mini E3 2.0 to Marlin folder.
3. Run platformio
4. Change to default_envs = STM32F103RC_btt_512K in platformio.ini
5. Build

**Expected behavior:**

Fix problem

I too get this warning message, many times within a single compiling:
```
Marlin\src\module\../inc/../HAL/./STM32F1/HAL.h:213:12: warning: 'int freeMemory()' defined but not used [-Wunused-function]
 static int freeMemory() {
```
SKR Mini E3 v1.2, `STM32F103RC_btt`/`STM32F103RC_btt_512K`
[Configuration.zip](https://github.com/MarlinFirmware/Marlin/files/5611004/Configuration.zip)

--- a/Marlin/src/HAL/STM32F1/HAL.h
+++ b/Marlin/src/HAL/STM32F1/HAL.h
@@ -202,17 +202,9 @@ extern "C" {
 
 extern "C" char* _sbrk(int incr);
 
-/*
-static int freeMemory() {
-  volatile int top;
-  top = (int)((char*)&top - reinterpret_cast<char*>(_sbrk(0)));
-  return top;
-}
-*/
-
-static int freeMemory() {
+static inline int freeMemory() {
   volatile char top;
-  return &top - reinterpret_cast<char*>(_sbrk(0));
+  return &top - _sbrk(0);
 }
 
 #if GCC_VERSION <= 50000

